src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/NativeLayoutVertexNode.cs

```diff
@@ -83,60 +83,135 @@ protected Vertex SetSavedVertex(Vertex value)
         }
     }
 
-    internal sealed class NativeLayoutMethodLdTokenVertexNode : NativeLayoutSavedVertexNode
+    internal abstract class NativeLayoutMethodEntryVertexNode : NativeLayoutSavedVertexNode
     {
-        private MethodDesc _method;
+        protected readonly MethodDesc _method;
+        private bool _createInstantiatedSignature;
+        private bool _saveEntryPoint;
         private NativeLayoutTypeSignatureVertexNode _containingTypeSig;
         private NativeLayoutMethodSignatureVertexNode _methodSig;
         private NativeLayoutTypeSignatureVertexNode[] _instantiationArgsSig;
 
-        protected override string GetName() => "NativeLayoutMethodLdTokenVertexNode_" + NodeFactory.NameMangler.GetMangledMethodName(_method);
-
-        public NativeLayoutMethodLdTokenVertexNode(NodeFactory factory, MethodDesc method)
+        public NativeLayoutMethodEntryVertexNode(NodeFactory factory, MethodDesc method, bool createInstantiatedSignature, bool saveEntryPoint)
```

Comment:
1. Instead of bools for saveEntryPoint, and _createInstantiatedSignature, could you use an enum? It would be easier to understand at sites which call this constructor.
2. Also, having the saveEntryPoint as an optional set of functionality feels odd. Why don't you have it implemented on the derived type?